### PR TITLE
[Launch API Docs] Implement the "Contents" right panel #12

### PR DESCRIPTION
**Pull request by rofe**
done:
- TOC creation
- simple styling
- testing
todo:
- advanced styling
- floating TOC

### CODE REVIEW

File: src/html.pre.js [hunk 1]
```diff
@@ -203,6 +203,35 @@ function computeNavPath(isDev, logger) {
...
+    if (child.match(hRegExp)) {
+      children[index] = child.replace(hRegExp, function(original, level, title) {
+        let id = index + '_' + encodeURIComponent(title);
+        // populate TOC
+        if (level < 4) {
+          toc.push('<li class="level-' + level + '"><a href="#' + id + '">' + title + '</a></li>');
+        }
+        // add id to <h*> tag
+        return '<h' + level + ' id="' + id + '">' +
+          title + '</h' + level + '>';
```
Comment (kptdobe) [inline src/html.pre.js]: String concatenation should be replaced with template strings. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Something like 

```
return `<h${level} id="${id}">$title</h${level}>`;
```

Comment (rofe) [reply src/html.pre.js]: Cool, thanks for the feedback! Will refactor and resubmit - with test ;)

Comment (rofe) [reply src/html.pre.js]: done

File: test/test.html.pre.js [hunk 2]
```diff
@@ -158,6 +158,32 @@ describe('Testing assembleEditUrl', () => {
...
     );
 
-    assert.deepEqual(output, 'https://github.com/test-owner/test-repo/edit/test-branch/test-file.md');
+    assert.equal(output, 'https://github.com/test-owner/test-repo/edit/test-branch/test-file.md');
+  });
+});
+
+describe('Testing createTOC', () => {
+  it('Create table of contents', () => {
+    let childrenBefore = [
```
Comment (kptdobe) [inline test/test.html.pre.js]: `let` or `const` ?

File: test/test.html.pre.js [hunk 3]
```diff
@@ -158,6 +158,32 @@ describe('Testing assembleEditUrl', () => {
...
+    ];
+    const expectedTOC = [
+      '<li class="level-2"><a href="#0_Heading%202">Heading 2</a></li>',
+      '<li class="level-3"><a href="#2_Heading%203">Heading 3</a></li>'
+    ];
+
+    const output = defaultPre.createTOC(childrenBefore, 3, loggerMock);
+
+    assert.deepEqual(output, expectedTOC);
+    assert.deepEqual(childrenBefore, childrenAfter);
```
Comment (kptdobe) [inline test/test.html.pre.js]: What about adding tests for edge cases, like empty children object and no heading in the list of children ? Usually, this is where it breaks ;)

File: test/test.html.pre.js (hunk 2 above)
Comment (rofe) [reply test/test.html.pre.js]: I believe `let` is correct because children actually get updated 😱

File: test/test.html.pre.js (hunk 3 above)
Comment (rofe) [reply test/test.html.pre.js]: Right, good point! Added tests for:
- different levels of nested headings
- no headings at all
- invalid input

**Comment by rofe** [comment]
added some eslint fixes
